### PR TITLE
[front] - refacto: connection management UI

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR refactors the connection management UI in the Dust front-end. The changes aim to simplify the look and feel, make available knowledge easier to understand, and allow for better management of connections. 

Key improvements include:
- Replacing the existing ContextItem.List with a DataTable component for better organization and searchability of connections.
- Adding a search functionality to quickly find specific connections.

The changes are part of the connection management efforts (cf. [Figma](https://www.figma.com/design/QOxHwppG64GtPocpDhS6Y7/Product?node-id=8487-51614&t=tULLJbvnBoSkntMf-1)).

New look:
Screenshot 2024-08-05 at 12 22 08

**References:**:
- https://www.figma.com/design/QOxHwppG64GtPocpDhS6Y7/Product?node-id=8487-51614&t=tULLJbvnBoSkntMf-1

## Risk

Breaking UX changes

## Deploy Plan

- Deploy to `front-edge`
- Deploy to `front`

### CODE REVIEW

File: front/pages/w/[wId]/builder/data-sources/managed.tsx [hunk 1]
```diff
@@ -166,6 +183,11 @@ export const getServerSideProps = withDefaultUserAuthRequirements<{
           connector: statusRes.value,
           fetchConnectorError: false,
           fetchConnectorErrorMessage: null,
+          editedByUser: mds.editedByUser?.imageUrl ?? null,
```
Comment (flvndvd) [inline front/pages/w/[wId]/builder/data-sources/managed.tsx:186]: Can we create a function to extract this duplicated code?

File: front/pages/w/[wId]/builder/data-sources/managed.tsx [hunk 2]
```diff
@@ -594,238 +657,284 @@ export default function DataSourcesView({
             </Hoverable>
           </ContentMessage>
         )}
-        <ContextItem.List>
-          {localIntegrations
```
Comment (flvndvd) [inline front/pages/w/[wId]/builder/data-sources/managed.tsx:598]: What's `localIntegrations` here?

File: front/pages/w/[wId]/builder/data-sources/managed.tsx [hunk 3]
```diff
@@ -594,238 +657,284 @@ export default function DataSourcesView({
...
+          filter={dataSourceSearch}
+          filterColumn={"name"}
+        />
       </Page.Vertical>
     </AppLayout>
   );
 }
+
+function getTableColumns() {
+  type Info = {
```
Comment (flvndvd) [inline front/pages/w/[wId]/builder/data-sources/managed.tsx:681]: Let's extract type from the function.

File: front/pages/w/[wId]/builder/data-sources/managed.tsx [hunk 4]
```diff
@@ -594,238 +657,284 @@ export default function DataSourcesView({
...
+              {original.comingSoonDialog}
+            </DataTable.Cell>
+          );
+        }
+      },
+    },
+  ];
+}
+
+function getTableRow(
```
Comment (flvndvd) [inline front/pages/w/[wId]/builder/data-sources/managed.tsx:809]: Let's avoid having that many parameters and use an object instead 🙏

File: front/pages/w/[wId]/builder/data-sources/managed.tsx (hunk 2 above)
Comment (JulesBelveze) [reply front/pages/w/[wId]/builder/data-sources/managed.tsx:598]: kept the previous naming. Will rename to `dataSourceIntegration` which is more explicit

File: front/pages/w/[wId]/builder/data-sources/managed.tsx [hunk 5]
```diff
@@ -158,6 +211,11 @@ export const getServerSideProps = withDefaultUserAuthRequirements<{
             connector: null,
             fetchConnectorError: true,
             fetchConnectorErrorMessage: statusRes.error.message,
+            editedByUser: getUserImageUrl(mds),
+            usage: await getDataSourceUsage({
```
Comment (flvndvd) [inline front/pages/w/[wId]/builder/data-sources/managed.tsx:215]: Is it okay to compute usage for all data sources like folders and websites? Isn't it too slow?

**Review by flvndvd — APPROVED** [review]

File: front/pages/w/[wId]/builder/data-sources/managed.tsx (hunk 5 above)
Comment (JulesBelveze) [reply front/pages/w/[wId]/builder/data-sources/managed.tsx:215]: Will definitely test it on front edge and create a filter if it is 👍🏼

Comment (JulesBelveze) [reply front/pages/w/[wId]/builder/data-sources/managed.tsx:215]: Looks alright, will add a ticket anyways 👍🏼

**Pull request merged**